src/setup-elixir.js

```diff
@@ -1,97 +1,356 @@
+        })
 
-  let installRebar = core.getInput('install-rebar')
-  installRebar = installRebar == null ? 'true' : installRebar
+      const rebar3BuildsListing = {
+        remote: 'https://api.github.com/repos/erlang/rebar3/releases',
+      }
+      const rebar3Spec = core.getInput('rebar3-version', {required: false})
+      maybeInstallRebar3(rebar3Spec, rebar3BuildsListing).catch((err) => {
+        throw err
+      })
+    })
+    .catch((err) => {
+      throw err
+    })
+}
 
-  console.log(`##[group]Installing OTP ${otpVersion} - built on ${osVersion}`)
-  await installOTP(otpVersion, osVersion)
-  console.log(`##[endgroup]`)
+function installOTP(otpSpec, osVersion, otpBuildsListing) {
+  return new Promise(async (resolve, reject) => {
+    await getOTPVersion(otpSpec, osVersion, otpBuildsListing)
+      .then(async (otpVersion) => {
+        console.log(
+          `##[group]Installing Erlang/OTP ${otpVersion} - built on ${osVersion}`,
+        )
+        await installer
+          .installOTP(osVersion, otpVersion)
+          .then(() => {
+            core.setOutput('otp-version', otpVersion)
+            console.log(`##[endgroup]`)
+            prependToPath(`${process.env.RUNNER_TEMP}/.setup-elixir/otp/bin`)
+            resolve(otpVersion)
+          })
+          .catch((err) => {
+            reject(err)
+          })
+      })
+      .catch((err) => {
+        reject(err)
+      })
+  })
+}
 
-  console.log(`##[group]Installing Elixir ${elixirVersion}`)
-  await installElixir(elixirVersion, otpMajor)
-  console.log(`##[endgroup]`)
+function maybeInstallElixir(elixirSpec, otpVersion, elixirBuildsListing) {
+  return new Promise(async (resolve, reject) => {
+    if (elixirSpec) {
+      await getElixirVersion(elixirSpec, otpVersion, elixirBuildsListing)
+        .then(async (elixirVersion) => {
+          console.log(`##[group]Installing Elixir ${elixirVersion}`)
+          await installer
+            .installElixir(elixirVersion)
+            .then(() => {
+              core.setOutput('elixir-version', elixirVersion)
+              console.log(`##[endgroup]`)
+              const matchersPath = path.join(__dirname, '..', '.github')
+              console.log(
+                `##[add-matcher]${path.join(matchersPath, 'elixir.json')}`,
+              )
+              prependToPath(
+                `${process.env.RUNNER_TEMP}/.setup-elixir/elixir/bin`,
+              )
+              resolve(true)
+            })
+            .catch((err) => {
+              reject(err)
+            })
+        })
+        .catch((err) => {
+          reject(err)
+        })
+    } else {
+      resolve(false)
+    }
+  })
+}
 
-  process.env.PATH = `${process.env.RUNNER_TEMP}/.setup-elixir/elixir/bin:${process.env.RUNNER_TEMP}/.setup-elixir/otp/bin:${process.env.PATH}`
+function mix(shouldMix, what) {
+  return new Promise(async (resolve, reject) => {
+    if (shouldMix) {
+      console.log(`##[group]Mixing local.${what}`)
```

Comment:
```suggestion
      console.log(`##[group]Running mix local.${what}`)
```

Reply:
Sure.

Reply:
This has now been pushed. Tests are still failing on purpose, because I wanna make sure consumption is eased, by:
1. having use fail fast, when expected,
2. not having unhandled exceptions.